### DISCUSSION TITLE
Validating path params

### DISCUSSION THREAD

**Discussion by jonahsnider** [adonisjs/core#2036] [Help] [answered]
Using `5.0.4-preview-rc`.

Wondering how to use the validator to validate path params. I saw an old forum post but I don't think the methods they described still work for the new validator.

I see `request.validate` has a description of

> Validate current request. The data is optional here, since request can pre-fill it for us

What data is this referring to? Can I pass in the custom params object?

**Comment by iagobruno** [comment, marked as answer]
To validate path params you can use the [`where` method](https://preview.adonisjs.com/guides/http/routing#regular-expression-matchers) in the route file to ensure that param matches a regular expression:

```ts
Route.get('/posts/:id', async ({ params }) => {
  // ...
}).where('id', /^[0-9]+$/) 👈
```

If the request path doesn't match param shape, the route will be ignored.

You can also write your own validation logic within route:

```ts
Route.get('/posts/:id', async ({ params, response }) => {
  if (!params.id.match(/^[0-9]+$/)) { 👈
    return response.status(400).send("Post id is not a number")
  }
  // ...
})
```

----

> What data is this referring to?

Data is the request body.

> Can I pass in the custom params object?

You can use the validator outside of HTTP requests, [as mentioned in the docs](https://preview.adonisjs.com/guides/validator/usage#standalone-usage).

**Reply by jonahsnider** [reply]
I think I'll use the validator standalone to check the params object. Thanks for the help!

**Reply by thetutlage** [reply]
You can validate it as follows:

```ts
request.validate({
  schema: schema.create({
    params: schema.object().members({
      
    })
  })
})
```
